### DISCUSSION TITLE
如果我想通过类似 CPM那样的方式导入是否可行

### DISCUSSION THREAD

**Discussion by Super1Windcloud** [xmake-io/xmake#5847] [General]
```CMake
CPMAddPackage(
    NAME spdlog
    GITHUB_REPOSITORY spdlog/spdlog
    VERSION 1.10.0
    OPTIONS "SPDLOG_BUILD_EXAMPLES OFF" # 禁用示例构建
)
or  
CPMAddPackage("gh:fmtlib/fmt#7.1.3") 
CPMAddPackage("gh:spdlog/spdlog@1.10.0")  
```

如何在xmake中, add_requires 是否支持 URL导入依赖

**Comment by star-hengxing** [comment]
cpm 是 fetchContent 加强版，和真正的包管理相比差得很多。
c/c++ 大部分库不可能一条 url 就能导入库完成构建，需要处理很多东西：
https://github.com/xmake-io/xmake-repo/blob/ed3d9fed41eaece613e205a0722ff2edf22170ee/packages/f/fmt/xmake.lua#L40-L71
